File: perl/examples/sort_days0.pl
#!/usr/bin/perl -w
%days = (Sunday => 0, Monday => 1, Tuesday => 2, Wednesday => 3,
         Thursday => 4, Friday => 5, Saturday => 6);
         
sub random_day {
	my @days = keys %days;
	return $days[rand @days];
}

sub compare_day {
	return $days{$a} <=> $days{$b};
}

push @random_days, random_day() foreach 1..5;
print "random_days=@random_days\n";
@sorted_days = sort compare_day @random_days;
print "sorted days=@sorted_days\n";

File: python/examples/sort_days0.py
#!/usr/bin/python
import random,operator
days = {'Sunday':0, 'Monday':1, 'Tuesday':2, 'Wednesday':3,
         'Thursday':4, 'Friday':5, 'Saturday':6}
         
def random_day():
	return str(random.choice(days.keys()))

def compare_day(day1, day2):
	return cmp(days[day1], days[day2])

random_days = [random_day() for x in xrange(0,5)]
print "random_days = ", ",".join(random_days)
sorted_days = sorted(random_days, cmp=compare_day)
print "sorted_days = ", ",".join(sorted_days)
